src/qibojit/backends/matrices.py

```diff
@@ -1,9 +1,18 @@
 from functools import cached_property  # pylint: disable=E0611
 
+import cupy as cp  # pylint: disable=import-error
```

Comment:
I think if you put this import here, we will need cupy a requirement of qibojit, otherwise it will fail to import anything from qibojit, even if it is numba related only. Currently it is not a requirement so that we don't force users who don't have a GPU and only use qibojit-numba to install cupy.

Alternatively (probably better) we can just move the import inside `def Unitary:`

Reply:
You are right. I forgot that cupy is not a dep. This explains also why all the imports are inside classes/functions.